daemon/algod/api/algod.oas2.json

```diff
@@ -1309,6 +1309,75 @@
       }
       ]
     },
+    "/v2/applications/{application-id}/boxes": {
+      "get": {
+        "description": "Given an application ID, it returns the box names of that application.",
+        "produces": [
+          "application/json"
+        ],
+        "schemes": [
+          "http"
+        ],
+        "summary": "Get boxes for a given application.",
+        "operationId": "GetApplicationBoxes",
+        "parameters": [
+          {
+            "type": "integer",
+            "description": "An application identifier",
+            "name": "application-id",
+            "in": "path",
+            "required": true
+          },
+          {
+            "type": "integer",
+            "description": "Number of box names to return. If max is not set, or max == 0, returns all box-names.",
+            "name": "max",
+            "in": "query"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "$ref": "#/responses/BoxesResponse"
+          },
+          "400": {
+            "description": "Bad Request",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "401": {
+            "description": "Invalid API Token",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "404": {
+            "description": "Application Not Found",
```

Comment:
@jannotti left this comment regarding `Application Not Found` errors: https://github.com/algorand/go-algorand/pull/4183#discussion_r910202402

Should we always return a `400` regardless of whether the application was not found (deleted) or the box was not found?

Reply:
I think we should simply return the empty set and not error. I don't think we should even be looking up the application during this operation, we just construct the box prefix and look for the boxes.  If the app does not exist, or if no boxes exists, the result is the same - an empty list.

Reply:
hmm, so I suppose keep other 200/400/401... but remove 404, right?